pages/conduct.md

```diff
@@ -0,0 +1,40 @@
+<script>{
+	"title": "jQuery Foundation Code of Conduct",
+	"pageTemplate": "page-conduct.php"
+}</script>
+
+Like the technical community as a whole, the jQuery team and community is made up of a mixture of professionals and volunteers from all over the world, working on every aspect of the mission - including mentorship, teaching, and connecting people.
+
+Diversity is one of our huge strengths, but it can also lead to communication issues and unhappiness. To that end, we have a few ground rules that we ask people to adhere to. This code applies equally to founders, mentors and those seeking help and guidance.
+
+This isn’t an exhaustive list of things that you can’t do. Rather, take it in the spirit in which it’s intended - a guide to make it easier to enrich all of us and the technical communities in which we participate.
+
+This code of conduct applies to all spaces managed by the jQuery Foundation and its projects. This includes IRC, the mailing lists, the issue tracker, jQuery events, and any other forums created by the project team which the community uses for communication. In addition, violations of this code outside these spaces may affect a person's ability to participate within them.
+
+If you believe someone is violating the code of conduct, we ask that you report it by emailing conduct@jquery.com. For more details please see our [Reporting Guidelines](https://jquery.org/conduct/reporting/)
+
+* **Be friendly and patient.**
+
+* **Be welcoming.** We strive to be a community that welcomes and supports people of all backgrounds and identities. This includes, but is not limited to members of any race, ethnicity, culture, national origin, colour, immigration status, social and economic class, educational level, sex, sexual orientation, gender identity and expression, age, size, family status, political belief, religion, and mental and physical ability.
+
+* **Be considerate.** Your work will be used by other people, and you in turn will depend on the work of others. Any decision you take will affect users and colleagues, and you should take those consequences into account when making decisions. Remember that we're a world-wide community, so you might not be communicating in someone else's primary language.
+
+* **Be respectful.** Not all of us will agree all the time, but disagreement is no excuse for poor behavior and poor manners. We might all experience some frustration now and then, but we cannot allow that frustration to turn into a personal attack. It’s important to remember that a community where people feel uncomfortable or threatened is not a productive one. Members of the jQuery community should be respectful when dealing with other members as well as with people outside the jQuery community.
```

Comment:
I think that ties into "being patient and friendly" from up above, which I would like to expand that section a bit more (you can find the comment thread it one of the outdated diffs).  Explain that that someone might act in an ignorant fashion because they are misinformed (or like being a jerk), but in order to maintain the type of community we would like to have, it's important for you to not act that way.  If it is a case that someone is uniformed, there are plenty of ways to explain the issue without coming back with a personal attack.

My personal rule tends to be if someone says something that pisses me off (and it's happened to me (not here) with some of the cases you've listed), I walk away from the problem for a little bit to give myself a chance to cool down.